python/lsst/afw/table/base/baseContinued.py

```diff
@@ -283,6 +283,24 @@ def asAstropy(self, cls=None, copy=False, unviewable="copy"):
             )
         return cls(columns, meta=meta, copy=False)
 
+    def __dir__(self):
+        # This custom dir is necessary due to the custom getattr below.
+        # Without it, not all of the methods available are returned with dir.
+        # See DM-7199
+        def recursive_get_class_dir(cls):
+            '''
+            Return a set containing the names of all methods
+            for a given class *and* all of its subclasses.
+            '''
+            result = set()
+            if cls.__bases__:
+                for subcls in cls.__bases__:
```

Comment:
Ah. I don't think this matters, because it's all going into a set, and ultimately lands in a sorted list that excludes duplicates. And I choose to take "exceptionally clever" as a compliment, largely on Russell's behalf.